packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,1152 @@
+    },
+  },
+  {
+    time: createRelativeDate(caseCreationDate, 5, 0).getTime(),
+    name: 'View',
+    value: {
+      content_id_key: 'sftitle',
+      content_id_value: 'Blaze pair with iPhone not working',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(secondSessionDate, -1, 0).getTime(),
+    name: 'Search',
+    value: {
+      query_expression: '',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(secondSessionDate, 1, 0).getTime(),
+    name: 'Click',
+    value: {
+      title: 'title',
+      uri_hash: 'KXñi9EWk38wnb1tt',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(secondSessionDate, 3, 0).getTime(),
+    name: 'Custom',
+    value: {
+      event_type: 'useless_event',
+      event_value: '',
+      origin_level_1: 'community-support',
+    },
+  },
+  {
+    time: createRelativeDate(secondSessionDate, 45, 0).getTime(),
+    name: 'Click',
+    value: {
+      title: 'title',
+      uri_hash: 'TtnKwc0Lo2GY9WAi',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(secondSessionDate, 60, 0).getTime(),
+    name: 'View',
+    value: {
+      content_id_key: 'sftitle',
+      content_id_value: 'Speedbit Charge 2 User Manual.pdf',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: firstSessionDate.getTime(),
+    name: 'Custom',
+    value: {
+      event_type: 'MySpeedbit App interfaceload',
+      event_value: '',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(firstSessionDate, 1, 0).getTime(),
+    name: 'Click',
+    value: {
+      title: 'title',
+      uri_hash: 'someHash',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(firstSessionDate, 5, 0).getTime(),
+    name: 'Search',
+    value: {
+      query_expression: 'Version 8.124',
+      origin_level_1: 'community-search',
+    },
+  },
+  {
+    time: createRelativeDate(firstSessionDate, 32, 0).getTime(),
+    name: 'Custom',
+    value: {
+      event_type: 'smartSnippetSuggestions',
+      event_value: 'expandSmartSnippetSuggestion',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(firstSessionDate, 34, 30).getTime(),
+    name: 'Custom',
+    value: {
+      event_type: 'smartSnippetSuggestions',
+      event_value: 'likeSmartSnippet',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(caseCreationDate, -6, 0).getTime(),
+    name: 'Custom',
+    value: {
+      event_type: 'MySpeedbit App interfaceload',
+      event_value: '',
+      origin_level_1: 'in-product-help',
+    },
+  },
+  {
+    time: createRelativeDate(caseCreationDate, -5, 0).getTime(),
+    name: 'Search',
+    value: {
+      query_expression: 'Blaze pair with iPhone not working',
+      origin_level_1: 'in-product-help',
+    },
+  },
```

Comment:
My initial feeling is that this sample data of fake actions is way to big.
Does every action here has a particular reason of being here? or you are just aiming for quantity here?

Cause I think our main focus is to create  a fake data in order to allow to make our tests robust by covering the main logic like you are doing but to also to cover the situations that were causing issues like the issues highlighted in the other PR comments. and to do this it doesn't necessary require a huge sample of data, we just need to pick the right timestamps on couple actions to allow us to confirm that the preprocessing script is working as expected.

Reply:
reduced the number of actions here